### PR TITLE
add ProChoice

### PR DESCRIPTION
**Pull request by julien-truffaut**

### CODE REVIEW

File: laws/src/main/scala/cats/laws/discipline/ProChoiceTests.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import cats.data.Xor
+import cats.functor.ProChoice
+import org.scalacheck.Arbitrary
+import org.scalacheck.Prop._
+
+trait ProChoiceTests[F[_, _]] extends ProfunctorTests[F] {
+  def laws: ProChoiceLaws[F]
+
+  def strong[A: Arbitrary, B: Arbitrary, C: Arbitrary, D: Arbitrary, E: Arbitrary, G: Arbitrary](implicit
```
Comment (non) [inline laws/src/main/scala/cats/laws/discipline/ProChoiceTests.scala]: Is this a name that failed to get changed?

**Comment by non** [comment]
Also, it doesn't look like the `ProChoice` tests are used anywhere. Could you add at least one use (i.e. for `Function1`)?

File: laws/src/main/scala/cats/laws/discipline/ProChoiceTests.scala (hunk 1 above)
Comment (julien-truffaut) [inline laws/src/main/scala/cats/laws/discipline/ProChoiceTests.scala]: good catch

**Comment by julien-truffaut** [comment]
sorry I didn't see that `Function1` had tests already.

**Comment by non** [comment]
Looks good to me :+1:

**Comment by stew** [comment]
Can we have an example of its usage somewhere? Like in a tut file?

**Comment by julien-truffaut** [comment]
sorry I haven't forgot about this. The only use I know is on the implementation of a `Prism`:
```haskell
type Prism s t a b = forAll p. (ProChoice p) => p a b => p s t 
```

I will try to find more examples

**Comment by julien-truffaut** [comment]
sorry I am really busy preparing move to the UK, I will probably not come back to this before months. It might be better to drop the PR, we can always introduce it later if there is a real interest.

**Comment by non** [comment]
@julien-truffaut No worries. I'm happy to leave this open but back-burner it.

**Comment by stew** [comment]
just created https://github.com/typelevel/cats/pull/1554/files which I believe makes this branch mergable.

**Comment by LukaJCB** [comment]
I think this can be closed as we already have `ArrowChoice` now :)

**Pull request closed without merging**